### PR TITLE
update/v2-updates

### PR DESCRIPTION
**Pull request by fivetran-sheringuyen**
Pull Request
**Are you a current Fivetran customer?** 

Fivetran Employee

**What change(s) does this PR introduce?** 

- Removed intermediate models and moved `is_most_recent_record` logic within the final models
- Updates to integration testing (default `*_using_search_terms` var is set to false)
- Updated to use identifier variables in integration tests
- Updated styling
- Updated README

**Did you update the CHANGELOG?** 

- [x] Yes
- [ ] No, package has not been released yet

**Does this PR introduce a breaking change?**

- [ ] Yes (please provide breaking change details below.)
- [x] No  (please provide an explanation as to how the change is non-breaking below.)
Not released yet

**Did you update the dbt_project.yml files with the version upgrade (please leverage standard semantic versioning)? (In both your main project and integration_tests)** 

- [ ] Yes
- [ ] Not released yet

**Is this PR in response to a previously created Bug or Feature Request**

- [ ] Yes, Issue/Feature [link bug/feature number here]
- [x] No 

**How did you test the PR changes?** 

 

- [x] CircleCi  
- [x] Local (please provide additional testing details below)

**Select which warehouse(s) were used to test the PR**
 

- [x] BigQuery
- [x] Redshift
- [x] Snowflake
- [x] Postgres
- [x] Databricks
- [ ] Other (provide details below)

**Provide an emoji that best describes your current mood**
 
:dancer:

**Feedback**

We are so excited you decided to contribute to the Fivetran community dbt package! We continue to work to improve the packages and would greatly appreciate your [feedback](https://www.surveymonkey.com/r/DQ7K7WW) on our existing dbt packages or what you'd like to see next.

### CODE REVIEW

File: models/apple_search_ads.yml
Comment (fivetran-joemarkiewicz) [inline models/apple_search_ads.yml:133]: It looks like the `match_type` field is not documented in this yml. Additionally, we are including the `created_at` field in the docs, but don't actually use it in this model.
image

File: models/apple_search_ads.yml
Comment (fivetran-joemarkiewicz) [inline models/apple_search_ads.yml:216]: It looks like the `match_type` field documentation was missed for this model.
image

File: packages.yml
Comment (fivetran-joemarkiewicz) [inline packages.yml:4]: Reminder to update before merge

File: models/apple_search_ads__ad_group_report.sql
Comment (fivetran-joemarkiewicz) [inline models/apple_search_ads__ad_group_report.sql:37]: I notice there are a number of other fields in the `history` staging models (such as `status`) that we don't bring through to this final cte. Is there a reason we don't bring them through to this final model?

Comment (fivetran-joemarkiewicz) [reply models/apple_search_ads__ad_group_report.sql:37]: Same question above for the other final models.

File: README.md
Comment (fivetran-joemarkiewicz) [inline README.md:17]: We should update this to be more specific to what this package is offering.

File: README.md
Comment (fivetran-joemarkiewicz) [inline README.md:26]: I think this needs to be updated for the Apple Search Ads report

File: README.md
Comment (fivetran-joemarkiewicz) [inline README.md:53]:
```suggestion
    version: [">=0.1.0", "<0.2.0"]
```

File: README.md
Comment (fivetran-joemarkiewicz) [inline README.md:108]: This link seems to be broken

File: README.md
Comment (fivetran-joemarkiewicz) [inline README.md:137]: There was no need for a DECISIONLOG.md in this package, right?

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-sheringuyen this is looking really good and so close to joining the dbt hub!

I just have a few comments and suggestions I would like you to take a look at before approving. Let me know if you have any questions.

File: models/apple_search_ads.yml
Comment (fivetran-sheringuyen) [reply models/apple_search_ads.yml:133]: Added & Removed respectively!

File: README.md
Comment (fivetran-sheringuyen) [reply README.md:108]: This link seems to be working for me 🤔 (C&p'd the link and went to the branch and clicked directly from the README)

File: README.md
Comment (fivetran-sheringuyen) [reply README.md:137]: Can confirm -- no `DECISIONLOG` needed in this package.

File: README.md
Comment (fivetran-joemarkiewicz) [reply README.md:108]: Same as the source. It seems my computer was just acting a funk when I reviewed this PR.

File: models/apple_search_ads__ad_group_report.sql
Comment (fivetran-sheringuyen) [reply models/apple_search_ads__ad_group_report.sql:37]: Just added a few more of the history fields into each report! Some fields I didn't bring in such as `bid_amount` / `currency` just because it's an additional currency metric that I think is nice to have as a part of the history tables however, am unsure how folks would use it in reports so I opted to keep those out of the models. But have added all `*_status` fields & `end_at` & `start_at` fields that could be useful in the end reports!

File: packages.yml
Comment (fivetran-joemarkiewicz) [inline packages.yml:4]: Reminder to switch before merge

**Review by fivetran-joemarkiewicz — APPROVED** [review]
LGTM! Thanks so much @fivetran-sheringuyen. This will be good to go once we regen the docs and switch the package dependency. 🏅

**Comment by fivetran-sheringuyen** [comment]
before merge notes:
- add keyword_id and keyword text to final search term model

**Pull request merged**
